homepage/models.py

```diff
@@ -12,26 +13,84 @@ class Course(models.Model):
     credit_points = models.SmallIntegerField(
         validators=[MinValueValidator(1),
                     MaxValueValidator(20)])  # the credit points assigned to this course by the college
-    syllabi = models.CharField(max_length=200)  # link to syllabi of the course, not for use, currently
+    syllabi = models.URLField(blank=True, null=True)  # link to syllabi of the course, not for use, currently
     avg_load = models.DecimalField(max_digits=6, decimal_places=5,
-                                   validators=[MinValueValidator(1), MaxValueValidator(5)])  # average course load
+                                   validators=[MinValueValidator(1), MaxValueValidator(5)],
+                                   blank=True, null=True)  # average course load, starts as null
     avg_rating = models.DecimalField(max_digits=6, decimal_places=5,
-                                     validators=[MinValueValidator(1), MaxValueValidator(5)])  # average course rating
-    num_of_raters = models.IntegerField(
-        validators=[MinValueValidator(0)])  # number of raters for course rating and course load
-    num_of_reviewers = models.IntegerField(
-        validators=[MinValueValidator(0)])  # number of reviewers
+                                     validators=[MinValueValidator(1), MaxValueValidator(5)],
+                                     blank=True, null=True)  # average course rating, starts as null
+    num_of_raters = models.IntegerField(validators=[MinValueValidator(0)],
+                                        default=0)  # number of raters for course rating and course load
+    num_of_reviewers = models.IntegerField(validators=[MinValueValidator(0)],
+                                           default=0)  # number of reviewers
 
     def __str__(self):
         return self.name
 
+    def clean(self):

```

Comment:
Maybe changes this method name?
its not clear to me what it does from the name. something like validate_course would be more clear IMO.

(if i understand correctly the purpose of this method is to validate that we can't add any course that has one of the characteristics you explained the the PR comment - since they need to be updated by the application logic and not by users)

Reply:
clean() is a method we get as a gift from django, which is applied if you choose to run someCourse.full_clean().
It is used to validate things that are not related to only one field in a Model instance.
More details [here](https://docs.djangoproject.com/en/3.2/ref/models/instances/#django.db.models.Model.clean)